hudi-client/src/main/java/org/apache/hudi/table/HoodieMergeOnReadTable.java

```diff
@@ -374,16 +374,12 @@ public void finalizeWrite(JavaSparkContext jsc, String instantTs, List<HoodieWri
             sf.location = new HoodieRecordLocation(FSUtils.getCommitTime(filename), FSUtils.getFileId(filename));
             sf.sizeBytes = getTotalFileSize(smallFileSlice);
             smallFileLocations.add(sf);
-            // Update the global small files list
-            smallFiles.add(sf);
```

Comment:
guess this is not needed anymore since we aggregate it into a map already.. (never liked this line. so lgtm)

Reply:
the UpsertPartitioner actually holds a global list small files for all partitions. since the work will be distributed to Task nodes, the method can't  the small file to global list anymore. So I moved the logic of updating that global list out side of the parallelism section.